modules/search-pipeline-common/src/main/java/org/opensearch/search/pipeline/common/README.md

```diff
@@ -0,0 +1,183 @@
+
+For the example field `DeleteFieldResponseProcessor`, this will look like:
+
+```
+@Override
+public SearchResponse processResponse(SearchRequest request, SearchResponse response) throws Exception {
+
+  boolean foundField = false;
+  SearchHit[] hits = response.getHits().getHits();
+  for (SearchHit hit : hits) {
+       
+    // Process each hit as desired 
+    
+    if (hit.hasSource()) {
+      // Change hit source if needed 
+    );
+
+    Map<String, Object> sourceAsMap = typeAndSourceMap.v2();
+    if (sourceAsMap.containsKey(field)) {
+      // Handle source as map 
+    }
+  }
+    
+  if (!foundField && !ignoreMissing) {
+      // Handle error scenarios 
+  }
+    
+  return response;
+}
+```
+
+4. Create a factory to parse processor-specific JSON configurations. These are used for constructing a processor instance. 
+
+In the `DeleteFieldResponseProcessor`, this would look something like:
+
+```
+public static final class Factory implements Processor.Factory<SearchResponseProcessor> {
+
+  /**
+   * Constructor for factory
+   */
+  Factory() {}
+
+  @Override
+  public DeleteFieldResponseProcessor create(
+      Map<String, Processor.Factory<SearchResponseProcessor>> processorFactories,
+      String tag,
+      String description,
+      boolean ignoreFailure,
+      Map<String, Object> config,
+      PipelineContext pipelineContext
+  ) throws Exception {
+      String field = ConfigurationUtils.readStringProperty(TYPE, tag, config, "field");
+      boolean ignoreMissing = ConfigurationUtils.readBooleanProperty(TYPE, tag, config, "ignore_missing", false);
+      return new DeleteFieldResponseProcessor(tag, description, ignoreFailure, field, ignoreMissing);
+  }
+}
+```
+
+In this example, we provide specific configurations for which field should be deleted and whether the processor should ignore attempts to remove a non-existent field.  
+
+5. Add the newly added search processor to the `SearchPieplineCommonModulePlugin` getter for the corresponding processor type.
+
+For the `DeleteFieldResponseProcessor`, you would modify the response processor getter to have:
+
+```
+@Override
+public Map<String, Processor.Factory<SearchResponseProcessor>> getResponseProcessors(Parameters parameters) { 
+  return Map.of(
+    RenameFieldResponseProcessor.TYPE, 
+    new RenameFieldResponseProcessor.Factory(),
+    DeleteFieldResponseProcessor.TYPE,
+    new DeleteFieldResponseProcessor.Factory()
+  );
+}
+```
+
+6. After creating a search processor, the processor is ready to be tested in a search pipeline. 
+
+To test your new search processor, you can make use of the test [`SearchPipelineCommonYamlTestSuiteIT`](../../../../../../../yamlRestTest/java/org/opensearch/search/pipeline/common). 
+
+Following the format of the YAML files in [`rest-api-spec.test.search_pipeline`](../../../../../../../yamlRestTest/resources/rest-api-spec/test/search_pipeline), you should be able to create your own YAML test file to exercise your new processor. 
+
```

Comment:
To run the tests, from the root of the OpenSearch repo, you can run `./gradlew :modules:search-pipeline-common:yamlRestTest`

Reply:
Added thanks!